### PR TITLE
Re-implement "comma fixes"

### PR DESCRIPTION
**Pull request by jugglinmike**
@rwaldron @caitp Writing something truly distinct was a little tough for this small patch, but the small size fortunately also makes originality less critical. I'm hoping that my discovery and correction of a previously-unreported bug lends some weight to the novelty of the new implementation.

### CODE REVIEW

File: src/jshint.js [hunk 1]
```diff
@@ -5227,7 +5244,7 @@ var JSHINT = (function() {
     if (!state.option.asi)
       nolinebreak(this);
 
-    if (state.tokens.next.id !== ";" && !state.tokens.next.reach &&
+    if (state.tokens.next.identifier &&
```
Comment (caitp) [inline src/jshint.js]: Does this change to ASI handling work properly if the next line begins with a pre-increment or pre-decrement?

**Comment by caitp** [comment]
other than the ASI handling question, lgtm

File: src/jshint.js (hunk 1 above)
Comment (rwaldron) [reply src/jshint.js]: @caitp I'm trying to think of an example, but maybe I'm misunderstanding you?

```js
var x = 0;
a: {
  break a
  --x
}
```

That doesn't seem right

Comment (caitp) [reply src/jshint.js]: I couldn't see that this was for stmt("break") from mobile, I guess it makes more sense now that I can see it

**Pull request merged**